easier_net/fit_easier_net.py

```diff
@@ -184,73 +184,75 @@ def main(args=sys.argv[1:]):
-            for fold_idx, init_idx in itertools.product(
-                range(num_folds), range(args.num_inits)
-            )
-        )
+    #     parallel = Parallel(n_jobs=args.n_jobs, verbose=True, pre_dispatch=args.n_jobs)
+    #     all_estimators = parallel(
+    #         delayed(_fit)(
+    #             base_estimator, #single sier net
+    #             x,
+    #             y,
+    #             train=fold_idx_dict[fold_idx]["train"],
+    #             max_iters=args.max_iters,
+    #             max_prox_iters=args.max_prox_iters,
+    #             seed=args.seed + num_folds * init_idx + fold_idx,
+    #         )
+    #         for fold_idx, init_idx in itertools.product(
+    #             range(num_folds), range(args.num_inits)
+    #         )
+    #     )
 
-        # Just printing things from the first fold
-        logging.info(f"sample estimator 0 fold 0")
-        all_estimators[0].net.get_net_struct()
+    #     # Just printing things from the first fold
+    #     logging.info(f"sample estimator 0 fold 0")
+    #     all_estimators[0].net.get_net_struct()
 
-        assert (num_folds * args.num_inits) == len(all_estimators)
+    #     assert (num_folds * args.num_inits) == len(all_estimators)
 
-        meta_state_dict = all_estimators[0].get_params()
-        meta_state_dict["state_dicts"] = [
-            [None for _ in range(num_folds)] for _ in range(args.num_inits)
-        ]
-        for (fold_idx, init_idx), estimator in zip(
-            itertools.product(range(num_folds), range(args.num_inits)), all_estimators
-        ):
-            meta_state_dict["state_dicts"][init_idx][
-                fold_idx
-            ] = estimator.net.state_dict()
-        torch.save(meta_state_dict, args.out_model_file)
-    else:
-        all_estimators = [
-            _fit(
-                base_estimator,
-                x,
-                y,
-                train=np.arange(x.shape[0]),
-                max_iters=args.max_iters,
-                max_prox_iters=args.max_prox_iters,
-                seed=args.seed + init_idx,
-            )
-            for init_idx in range(args.num_inits)
-        ]
-        meta_state_dict = all_estimators[0].get_params()
-        meta_state_dict["state_dicts"] = [
-            estimator.net.state_dict() for estimator in all_estimators
-        ]
-        torch.save(meta_state_dict, args.out_model_file)
+    #     meta_state_dict = all_estimators[0].get_params()
+    #     meta_state_dict["state_dicts"] = [
+    #         [None for _ in range(num_folds)] for _ in range(args.num_inits)
+    #     ]
+    #     for (fold_idx, init_idx), estimator in zip(
+    #         itertools.product(range(num_folds), range(args.num_inits)), all_estimators
+    #     ):
+    #         meta_state_dict["state_dicts"][init_idx][
+    #             fold_idx
+    #         ] = estimator.net.state_dict()
+    #     torch.save(meta_state_dict, args.out_model_file)
+    # else:
+    all_estimators = [
```

Comment:
dont need to do this loop anymore because you have an `EasierNetEstimator`. should be able to fit an easier net the usual sklearn way